### PR TITLE
Stratos Metering Service Dashboard

### PR DESCRIPTION
**Pull request by Thanu**
This p/r includes publishing data for metering service and creating custom dashboard for visualize the data.
Following changes are included here:
1. Refactoring of thrift statistics publisher classes.
2. Modification of thrift-client-config.xml to enable data publishing for DAS.
3. Refactoring of streams and publisher classes to publish metering data to DAS.
4. Addition of Stratos Metering Service dashboard Composite  Application with spark udf as das extension.

### CODE REVIEW

File: components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/statistics/publisher/DASMemberInformationPublisher.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * MemberInfoPublisher to publish member information/metadata to DAS.
+ */
+public class DASMemberInformationPublisher extends ThriftStatisticsPublisher implements MemberInformationPublisher {
+
+    private static final Log log = LogFactory.getLog(DASMemberInformationPublisher.class);
+
+    private static final String DATA_STREAM_NAME = "member_info";
+    private static final String VERSION = "1.0.0";
+    private static final String DAS_THRIFT_CLIENT_NAME = "das";
+    private static final String NULL_VALUE = "Value Not Found";
```
Comment (imesh) [inline components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/statistics/publisher/DASMemberInformationPublisher.java]: May be we can rename this constant to VALUE_NOT_FOUND.

File: extensions/das/metering-service/capps/jaggery-files/clusters.jag [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<%
+/*
+ *
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/clusters.jag]: May be we can remove the additional lines above.

File: extensions/das/metering-service/capps/jaggery-files/applications.jag [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<%
+/*
+ *
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/applications.jag]: May be we can remove the additional lines above.

File: extensions/das/metering-service/capps/jaggery-files/member-count.jag [hunk 4]
```diff
@@ -0,0 +1,99 @@
+<%
+/*
+ *
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/member-count.jag]: May be we can remove the additional lines above.

File: extensions/das/metering-service/capps/jaggery-files/member-info.jag [hunk 5]
```diff
@@ -0,0 +1,71 @@
+<%
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/member-info.jag]: May be we can remove the additional lines above.

File: extensions/das/metering-service/capps/jaggery-files/member-info.jag [hunk 6]
```diff
@@ -0,0 +1,71 @@
+<%
+/*
+ *
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/member-info.jag]: May be we can remove the additional lines above.

File: extensions/das/metering-service/capps/jaggery-files/member-status.jag [hunk 7]
```diff
@@ -0,0 +1,91 @@
+<%
+/*
+ *
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/member-status.jag]: May be we can remove the additional lines above.

File: extensions/das/metering-service/capps/jaggery-files/schema.jag [hunk 8]
```diff
@@ -0,0 +1,64 @@
+<%
+/*
+ *
```
Comment (imesh) [inline extensions/das/metering-service/capps/jaggery-files/schema.jag]: May be we can remove the additional lines above.

Comment (Thanu) [inline extensions/das/metering-service/capps/jaggery-files/schema.jag]: Fixed with commit 79e76ddc87db226daa441bab05fcb31f9132e00f.

File: components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/statistics/publisher/DASMemberInformationPublisher.java (hunk 1 above)
Comment (Thanu) [inline components/org.apache.stratos.cloud.controller/src/main/java/org/apache/stratos/cloud/controller/statistics/publisher/DASMemberInformationPublisher.java]: Fixed with commit e990cdc1fba32a345b0076ae0536a71fdfda442e.

File: extensions/das/metering-service/capps/jaggery-files/clusters.jag (hunk 2 above)
Comment (Thanu) [inline extensions/das/metering-service/capps/jaggery-files/clusters.jag]: Fixed with commit 79e76ddc87db226daa441bab05fcb31f9132e00f.

File: extensions/das/metering-service/capps/jaggery-files/applications.jag (hunk 3 above)
Comment (Thanu) [inline extensions/das/metering-service/capps/jaggery-files/applications.jag]: Fixed with commit 79e76ddc87db226daa441bab05fcb31f9132e00f.

File: extensions/das/metering-service/capps/jaggery-files/member-count.jag (hunk 4 above)
Comment (Thanu) [inline extensions/das/metering-service/capps/jaggery-files/member-count.jag]: Fixed with commit 79e76ddc87db226daa441bab05fcb31f9132e00f.

File: extensions/das/metering-service/capps/jaggery-files/member-info.jag (hunk 6 above)
Comment (Thanu) [inline extensions/das/metering-service/capps/jaggery-files/member-info.jag]: Fixed with commit 79e76ddc87db226daa441bab05fcb31f9132e00f.

**Comment by imesh** [comment]
Thanks Thanuja for the contribution, I have now merged this to stratos-4.1.x branch. Please close the pull request.

**Comment by Thanu** [comment]
Closing this P/R as it is merged.

**Pull request closed without merging**